proposals/interceptors-issues-2024-01.md

```diff
@@ -0,0 +1,147 @@
+# Open issues for interceptors
+
+## File path portability
+
+One of our design goals with interceptors was to have a particular form of "portability": essentially, we should be able to run the source generators for a project, commit the results to disk, then remove the source generators from the project and be able to build in a variety of environments. Note that this is different from *determinism*--we don't mind if the resulting assembly differs in things like layout, etc. in each build environment, but want it to be the case that if an interceptor declaration functions in one environment, then it also functions in a different environment, assuming all the sources and dependencies are fully available in each environment.
+
+However, we currently require absolute paths for `InterceptsLocationAttribute`, with pathmap substitution performed if present. This doesn't work across local builds and CI because only CI will actually pass `/pathmap` during the build. The local build prefers to preserve local paths (e.g. in the PDB) so that the local versions of the sources can be discovered easily during debugging. The local and CI environments will have different answers for what they expect the paths to be, and the InterceptsLocation will have errors in one of the two environments.
+
+Having looked at several strategies to address here, including adjusting when the build system passes `/pathmap` to the compiler, it seems the most straightforward way is to permit relative paths. This is a strategy which is already proven out by `#line` directives.
```

Comment:
I would end here with the reason we didn't permit relative paths in preview is that generated files don't have a known path. It's an undefined value. Hence relative paths didn't work as a solution.

Reply:
Do we intend for generated files to have a known path moving forward? Further down, I read:

> When a relative path appears in an `[InterceptsLocation]`, we will resolve it relative to the path of the containing file,

Does "containing file" here refer to the generated file or the file containing the original invocation?